isis/src/base/objs/CSMCamera/CSMCamera.cpp

```diff
@@ -88,46 +88,50 @@ namespace Isis {
   }
 
 
+  // TODO make this an iterative technique that updates the locus each iteration
   bool CSMCamera::SetImage(const double sample, const double line) {
+    csm::ImageCoord imagePt;
+    isisToCsmPixel(p_alphaCube->AlphaLine(line), p_alphaCube->AlphaSample(sample), imagePt);
+    double achievedPrecision = 0;
+    csm::WarningList warnings;
 
-// move to protected?
-    p_childSample = sample;
-    p_childLine = line;
-//    p_pointComputed = true;
-
-//    if (p_projection == NULL || p_ignoreProjection) {
-        double parentSample = p_alphaCube->AlphaSample(sample);
-        double parentLine = p_alphaCube->AlphaLine(line);
-//        bool success = false;
-
-        double height = 10.0;
-        csm::ImageCoord imagePt(parentLine, parentSample);
-
-        // do image to ground with csm
-        csm::EcefCoord result = m_model->imageToGround(imagePt, height);
-
-        // Set X, Y, Z in surface point
-        double naifValues[3] = {result.x, result.y, result.z};
-        target()->shape()->surfaceIntersection()->FromNaifArray(naifValues);
-
-        // check set of coordinate:
-        double test[3];
-        Coordinate(test);
-        // std::cout << "TEST:  " << test[0] << ", "  << test[1] << ", " << test[2] << std::endl;
-        // std::cout << "UniversalLatitude: " << UniversalLatitude() << std::endl;
+    csm::EcefLocus imageLocus;
+    try {
+      imageLocus = m_model->imageToRemoteImagingLocus(imagePt,
+                                                      0.001,
+                                                      &achievedPrecision,
+                                                      &warnings);
+    }
+    catch (csm::Error &e) {
+      return false;
+    }
 
-            // get a lat, lon and store in variables
-        // (1) how to get lat/lon
-        // (2) which variables to store in
+    // Check for issues on the CSM end
+    if (achievedPrecision > 0.001) {
+      return false;
+    }
+    if (!warnings.empty()) {
+      for (csm::Warning warning : warnings) {
+        if (warning.getWarning() == csm::Warning::IMAGE_COORD_OUT_OF_BOUNDS){
+          return false;
+        }
+      }
+    }
 
-        // fill in whatever stuff ISIS needs from this
-//    }
-//    else {
-        // handle projected case
-//    }
+    // ISIS sensors work in Kilometers, CSM works in meters
+    std::vector<double> obsPosition = {imageLocus.point.x / 1000.0,
+                                       imageLocus.point.y / 1000.0,
+                                       imageLocus.point.z / 1000.0};
```

Comment:
I would like to put this through the csm->ISIS conversion but I'm concerned about it being slow and it still requires making a surface point and specifying that it's in meters. So, it's still prone to errors for unit conversions.